### ISSUE TITLE
Multiple collapses that shouldn't happen in one move

### ISSUE THREAD

**Issue by Naruyoko** [Reinhardt-C/2048#1]
Example:
```
. . . .
. . . .
. . . .
. 4 2 2
Move right (ignoring spawn):
. . . .
. . . .
. . . .
. . . 8
Expected (ignoring spawn):
. . . .
. . . .
. . . .
. . 4 4
```

**Comment by Reinhardt-C** [comment]
Yeah, I noticed this but couldn't think of how to fix it, I think I have an idea. If I give a cell a property called `justMerged` or something, which is removed whenever a move begins, but prevents a move from occuring, that could fix the issue. The way the cells are ordered in the array (I store as an array rather than a grid to be more dynamic, but kinda causes this issue) determines whether this issue occurs or one where you have 2 2 4 and it collapses to 8, I've got the order in the way to avoid this, but I do have to prevent this from occuring. Thanks!
